drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -281,6 +283,27 @@ static int mtd_spi_read_jedec_id(const mtd_spi_nor_t *dev, mtd_jedec_id_t *out)
     return status;
 }
 
+/**
+ * @internal
+ * @brief Get Flash capacity based on JEDEC ID
+ *
+ * @note The way the capacity is encoded differs between vendors.
+ *       This formula has been tested with flash chips from Adesto,
+ *       ISSI, Micron and Spansion, but it might not cover all cases.
+ *       Please extend the function if necessary.
+ */
+static uint32_t mtd_spi_nor_get_size(const mtd_jedec_id_t *id)
+{
+    /* AT25SFxxx uses 5 lower bits of device ID 1 for density */
+    if ((id->manuf == SPI_NOR_JEDEC_ADESTO) &&
+        (id->device[0] & 0xE0) == 0x80) {
```

Comment:
This should also test if the bank where this ID was found matches the bank assigned to Adesto I think

Reply:
What do you mean by bank assigned to Adesto?

Reply:
btw I think we can drop the family check: I picked a [random Adesto flash](https://www.mouser.de/datasheet/2/590/DS-AT25PE16_143-1149783.pdf) and that also uses the same size coding.

The GigaDevices Flash on the Serpente board uses the more common Device ID2 encoding. 

(Which also revealed an error in my previous patch: It encodes the size in bits, not bytes. My old code used `MBIT` everywhere but when rewriting it for RIOT it tried to be clever…)

Reply:
Hmm looks like [newer Adesto chips](https://www.adestotech.com/wp-content/uploads/AT25SF128A.pdf) also use the ID2 coding - looks like that got standardized. 

![image](https://user-images.githubusercontent.com/1301112/82350443-a8706280-99fb-11ea-8112-7f165a3046ec.png)

I'll see if there is a way to identify the ones that use the old scheme.

Reply:
On these older chips, device ID 2 seems to always be 0, 1 or two - it encodes the product version and "Sub Code".

I think if we check for that, we should be good.

Reply:
> What do you mean by bank assigned to Adesto?

From JEP106AV, the jedec ID's are 8 bit IDs. To extend the number of available IDs, multiple banks (addresses) are used. The ISSI ID (0x9D) is on bank 2, Lucent (AT&T) has the same ID, but then on bank 1.

Reply:
Oh I didn't know that - I guess my little enum is then incomplete.

Reply:
Hm thinking about it I wonder if it makes sense to carry IDs that we don't use when the list is so much longer.

Initially I thought this would be a good marker to indicate where that density scheme works, but since that appears to be the standard and only Adesto being an outliner, I think we could just drop it.

Reply:
There is probably not a way to automatically detect which sheme is used right?

Reply:
I haven't found a better way, but I haven't looked very hard.